staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go

```diff
@@ -367,70 +372,47 @@ func (o *DeleteOptions) RunDelete(f cmdutil.Factory) error {
 func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
 	found := 0
 	if o.IgnoreNotFound {
-		r = r.IgnoreErrors(errors.IsNotFound)
+		r = r.IgnoreErrors(apierrors.IsNotFound)
+	}
+	if o.Concurrent {
+		o.WarningPrinter.Print("concurrent deletion of resources, not guaranteed to be in order")
 	}
 	warnClusterScope := o.WarnClusterScope
 	deletedInfos := []*resource.Info{}
 	uidMap := cmdwait.UIDMap{}
-	err := r.Visit(func(info *resource.Info, err error) error {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var errMutex sync.Mutex
+	var errorList []error
+	visitFunc := func(info *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
-
-		if o.Interactive {
-			if _, ok := o.previewResourceMap[cmdwait.ResourceLocation{
-				GroupResource: info.Mapping.Resource.GroupResource(),
-				Namespace:     info.Namespace,
-				Name:          info.Name,
-			}]; !ok {
-				// resource not in the list of previewed resources based on resourceLocation
-				return nil
-			}
-		}
-
-		deletedInfos = append(deletedInfos, info)
-		found++
-
-		options := &metav1.DeleteOptions{}
-		if o.GracePeriod >= 0 {
-			options = metav1.NewDeleteOptions(int64(o.GracePeriod))
-		}
-		options.PropagationPolicy = &o.CascadingStrategy
-
-		if warnClusterScope && info.Mapping.Scope.Name() == meta.RESTScopeNameRoot {
-			o.WarningPrinter.Print("deleting cluster-scoped resources, not scoped to the provided namespace")
-			warnClusterScope = false
-		}
-
-		if o.DryRunStrategy == cmdutil.DryRunClient {
-			if !o.Quiet {
-				o.PrintObj(info)
+		if o.Concurrent {
+			wg.Add(1)
+			go func() {
```

Comment:
I'm curious about one thing: If we only use one goroutine and use wait to wait for the return result, is there any difference from the original implementation?